system-contracts/test/SystemContext.spec.ts

```diff
@@ -0,0 +1,400 @@
+      expect(result).to.be.not.equal(gasPriceBefore);
+    });
+  });
+
+  describe("getBatchNumberAndTimestamp", async () => {
+    it("should get batch number and timestamp", async () => {
+      const result = await systemContext.getBatchNumberAndTimestamp();
+      expect(result.batchNumber).to.be.equal(0);
+      expect(result.batchTimestamp).to.be.equal(0);
+    });
+
+    it("should get changed batch data", async () => {
+      await systemContext.connect(bootloaderAccount).unsafeOverrideBatch(222, 111, 333);
+      const batchDataAfterChanges = await systemContext.getBatchNumberAndTimestamp();
+      const baseFee = await systemContext.baseFee();
+      expect(batchDataAfterChanges.batchNumber).to.be.equal(111);
+      expect(batchDataAfterChanges.batchTimestamp).to.be.equal(222);
+      expect(baseFee).to.be.equal(333);
+      await systemContext.connect(bootloaderAccount).unsafeOverrideBatch(0, 0, 0);
+      const batchDataRestored = await systemContext.getBatchNumberAndTimestamp();
+      const baseFeeRestored = await systemContext.baseFee();
+      expect(batchDataRestored.batchNumber).to.be.equal(0);
+      expect(batchDataRestored.batchTimestamp).to.be.equal(0);
+      expect(baseFeeRestored).to.be.equal(0);
+    });
+  });
+
+  describe("setNewBatch", async () => {
+    it("should get hash of the given batch", async () => {
+      const batchData = await systemContext.getBatchNumberAndTimestamp();
+      const result = await systemContext.getBatchHash(batchData.batchNumber);
+      expect(result).to.equal(ethers.constants.HashZero);
+    });
+
+    it("should revert not called by bootlader", async () => {
+      const batchData = await systemContext.getBatchNumberAndTimestamp();
+      const batchHash = await systemContext.getBatchHash(batchData.batchNumber);
+      await expect(
+        systemContext.setNewBatch(batchHash, batchData.batchTimestamp.add(1), batchData.batchNumber.add(1), 1)
+      ).to.be.rejectedWith("Callable only by the bootloader");
+    });
+
+    it("should revert timestamp should be incremental", async () => {
+      const batchData = await systemContext.getBatchNumberAndTimestamp();
+      const batchHash = await systemContext.getBatchHash(batchData.batchNumber);
+      await expect(
+        systemContext
+          .connect(bootloaderAccount)
+          .setNewBatch(batchHash, batchData.batchTimestamp, batchData.batchNumber.add(1), 1)
+      ).to.be.rejectedWith("Timestamps should be incremental");
+    });
+
+    it("should revert wrong block number", async () => {
+      const batchData = await systemContext.getBatchNumberAndTimestamp();
+      const batchHash = await systemContext.getBatchHash(batchData.batchNumber);
+      await expect(
+        systemContext
+          .connect(bootloaderAccount)
+          .setNewBatch(batchHash, batchData.batchTimestamp.add(1), batchData.batchNumber, 1)
+      ).to.be.rejectedWith("The provided block number is not correct");
```

Comment:
considering that it goes into `dev` (i.e. smart contract changes are permitted), could you please change this error both here and in the system context contract to `The provided batch number is not correct`

Reply:
@matzayonc are you sure it has been done? I don't see it changed anywhere